File: project-{topic}-advanced/backend/src/entities/Role.ts
```typescript
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from 'typeorm';
import { User } from './User';

export enum UserRole {
  USER = 'user',
  ADMIN = 'admin',
  GUEST = 'guest', // Example for a less privileged role
}

@Entity('roles')
export class Role {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'enum', enum: UserRole, default: UserRole.USER, unique: true })
  name: UserRole;

  @Column({ nullable: true })
  description: string;

  @OneToMany(() => User, user => user.role)
  users: User[];
}
```

File: project-{topic}-optimized/src/main/java/com/alx/devops/model/Role.java
package com.alx.devops.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

/**
 * Represents a user role in the system.
 * Used for authorization (e.g., ADMIN, USER).
 */
@Entity
@Table(name = "roles")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(length = 20, unique = true, nullable = false)
    private RoleName name;

    public Role(RoleName name) {
        this.name = name;
    }
}

enum RoleName {
    ROLE_USER,
    ROLE_ADMIN
}